runtime/list.go

```diff
@@ -170,6 +170,38 @@ func listCount(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
 	return seqCount(f, args[0], args[1])
 }
 
+func listRemove(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if raised := checkMethodArgs(f, "remove", args, ListType, ObjectType); raised != nil {
+		return nil, raised
+	}
+	v := args[1]
+	i := -1
+	pred := func(o *Object) (bool, *BaseException) {
+		i++
+		eq, raised := Eq(f, o, v)
+		if raised != nil {
+			return false, raised
+		}
+		ret, raised := IsTrue(f, eq)
+		if raised != nil {
+			return false, raised
+		}
+		return ret, nil
+	}
+	foundEqItem, raised := seqFindFirst(f, args[0], pred)
+	if raised != nil {
+		return nil, raised
+	}
+	if !foundEqItem {
+		return nil, f.RaiseType(ValueErrorType, "list.remove(x): x not in list")
+	}
+	l := toListUnsafe(args[0])
+	l.mutex.Lock()
```

Comment:
You need to lock the list during the seqFindFirst operation as well, otherwise the list could change between determining the item index and attempting to remove it.